core/src/main/java/com/novoda/noplayer/mediaplayer/forwarder/CompletionInfoForwarder.java

```diff
@@ -0,0 +1,24 @@
+package com.novoda.noplayer.mediaplayer.forwarder;
+
+import android.media.MediaPlayer;
+
+import com.novoda.noplayer.listeners.InfoListeners;
+
+import java.util.HashMap;
+
+class CompletionInfoForwarder implements MediaPlayer.OnCompletionListener {
+
+    private final InfoListeners infoListeners;
+
+    public CompletionInfoForwarder(InfoListeners infoListeners) {
+        this.infoListeners = infoListeners;
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        HashMap<String, String> keyValuePairs = new HashMap<>();
+        keyValuePairs.put("mp", String.valueOf(mp));
```

Comment:
Well, all of the info listeners are used for logging really. Client's shouldn't really rely on it for anything else.